src/md.cc

```diff
@@ -432,10 +457,44 @@ void MGmol<OrbitalsType>::md(OrbitalsType** orbitals, Ions& ions)
         int retval              = 0;
         bool small_move         = true;
         bool last_move_is_small = true;
-        do
+
+        bool force_on_ions = true;
+#ifdef MGMOL_HAS_LIBROM
+        // Ions for ROM
+        Mesh* mymesh            = Mesh::instance();
+        const pb::Grid& mygrid  = mymesh->grid();
+        const double lattice[3] = { mygrid.ll(0), mygrid.ll(1), mygrid.ll(2) };
+        std::vector<double> positions;
+        std::vector<short> anumbers;
+        getAtomicPositions(positions);
+        getAtomicNumbers(anumbers);
+        Ions ROM_ions(positions, anumbers, lattice, ions_->getSpecies());
+
+        // Pinned H2O 3 DOF
+        std::vector<std::string> names;
+        PinnedH2O H2O_molecule;
+        if (ct.getROMOptions().rom_stage == ROMStage::ONLINE_PINNED_H2O_3DOF)
+        {
+            if (onpe0) os_ << "Rotate Pinned H2O molecule in timestep " << mdstep << std::endl;
+            H2O_molecule.rotate(positions, anumbers);
+            H2O_molecule.print(os_);
+            ROM_ions.setPositions(positions, anumbers);
```

Comment:
I would create ROM_ions here instead of earlier, and not use setPositions

Reply:
Thanks for the suggestion @jeanlucf22. I think `ROM_ions` need to be declared outside (before) the if-statement on Line 476, since it will be used in Line 567. I am not aware of any ways to declare `ROM_ions` first and then use that constructor later to initialize it inside the if-statement.

Reply:
declare a pointer to it and allocate it inside the if-statement

Reply:
Done